### DISCUSSION TITLE
Why is onChange prop required in RadioGroup?

### DISCUSSION THREAD

**Discussion by arodriguezcortes** [twilio-labs/paste#2358] [General] [answered]
Hi Paste team. I'd like to implement an uncontrolled version of the Radio input using Paste, but seems that the RadioGroup component forces us to add an onChange prop (check [here](https://github.com/twilio-labs/paste/blob/e971512c5570f11132e6893a2ab635e6568c4201/packages/paste-core/components/radio-group/src/RadioGroup.tsx#L9)). I've been analyzing this component and I'm not able to find the reason for why is this prop required. For example, on [line 31](https://github.com/twilio-labs/paste/blob/e971512c5570f11132e6893a2ab635e6568c4201/packages/paste-core/components/radio-group/src/RadioGroup.tsx#L31), you are checking if the prop is null or not.

If this component is meant to be a controlled component, why is only the onChange prop required and not the value prop?. If it is meant to be used as either a controlled or uncontrolled component, shouldn't we change the onChange prop to optional?

Thanks in advance.

**Comment by TheSisb** [comment, marked as answer]
Hi @arodriguezcortes,

Good catch, our RadioGroup is currently controlled only. I have made a ticket in our backlog to review the component and make necessary fixes: https://issues.corp.twilio.com/browse/DSYS-3311.

We'll report back when we complete the work.

**Reply by arodriguezcortes** [reply]
Thanks @TheSisb !

**Reply by TheSisb** [reply]
Hey @arodriguezcortes,

[This PR](https://github.com/twilio-labs/paste/pull/2375) implements a fix and will be released to NPM over the next few days. Thanks again for the bug report!
